extensions/amp-story/1.0/amp-story-embedded-component.js

```diff
@@ -536,37 +574,152 @@ export class AmpStoryEmbeddedComponent {
+   * preparation for its animation. It resizes it to its full-screen size, and
+   * scales it down to match size set by publisher, adding negative margins so
+   * that content around stays put.
+   * @param {!Element} pageEl
+   * @param {!Element} element
+   * @param {!../../../src/service/resources-impl.Resources} resources
+   */
+  static prepareForAnimation(pageEl, element, resources) {
+    let elId = null;
+
+    // When a window resize happens, we must reset the styles and prepare the
+    // animation again.
+    if (element.hasAttribute(EMBED_ID_ATTRIBUTE_NAME)) {
+      elId = element.getAttribute(EMBED_ID_ATTRIBUTE_NAME);
+      embedStyleEls[elId].textContent = '';
+      embedStyleEls[elId][AMP_EMBED_DATA] = {};
+    }
 
-          state.translateY = centeredTop - topOffset;
-          state.translateX = leftOffset - centeredLeft;
+    const state = {};
+    resources.measureMutateElement(element,
+        /** measure */
+        () => {
+          const pageRect = pageEl./*OK*/getBoundingClientRect();
+          const elRect = element./*OK*/getBoundingClientRect();
+
+          if (elRect.width >= elRect.height) {
+            state.newWidth = pageRect.width;
+            state.scaleFactor = elRect.width / state.newWidth;
+            state.newHeight = elRect.height / elRect.width * state.newWidth;
+          } else {
+            const maxHeight = pageRect.height - VERTICAL_PADDING;
+            state.newWidth = Math.min(
+                elRect.width / elRect.height * maxHeight, pageRect.width);
+            state.newHeight = elRect.height / elRect.width * state.newWidth;
+            state.scaleFactor = elRect.height / state.newHeight;
+          }
+
+          state.verticalMargin =
+            (-1 * ((state.newHeight - elRect.height) / 2));
+          state.horizontalMargin =
+            (-1 * ((state.newWidth - elRect.width) / 2));
         },
         /** mutate */
         () => {
-          target.classList.add('i-amphtml-animate-expand-in');
-          setImportantStyles(dev().assertElement(target),
-              {
-                transform: `translate3d(${state.translateX}px,
-                    ${state.translateY}px, 0)`,
-              });
+          element.classList.add('i-amphtml-embedded-component');
+
+          elId = elId ? elId : ++embedIds;
+
+          const styleTextContent = `[${EMBED_ID_ATTRIBUTE_NAME}="${elId}"] {
+            width: ${px(state.newWidth)} !important;
+            height: ${px(state.newHeight)} !important;
+            transform: scale(${state.scaleFactor}) !important;
+            margin: ${state.verticalMargin}px ${state.horizontalMargin}px
+                !important;
+            }`;
+
+          if (element.hasAttribute(EMBED_ID_ATTRIBUTE_NAME)) { // Embed style already exists.
+            embedStyleEls[elId].textContent = styleTextContent;
+          } else {
+            const html = htmlFor(pageEl);
+            const embedStyleEl = html`<style></style>`;
```

Comment:
You can do document.createElement('style')

Reply:
I could, but since this is a static method I don't have access to document. Do you have a strong preference?